src/PSDMatrices.jl

```diff
@@ -17,6 +17,9 @@ inv(M::PSDMatrix) = PSDMatrix(inv(M.R'))
 \(A::PSDMatrix, B::AbstractVecOrMat) = A.R \ (A.R' \ B)
 /(B::AbstractVecOrMat, A::PSDMatrix) = B / A.R / A.R'
 copy(M::PSDMatrix) = PSDMatrix(copy(M.R))
+similar(M::PSDMatrix, element_type::Type{T}=eltype(M)) where {T,N} =
```

Comment:
I'll be honest, I dont understand this `where {T, N} = ...` bit (especially the `N`). But it seems to be solid :)

Reply:
Good catch actually. To be honest, originally I had something like
```julia
similar(M::PSDMatrix) = PSDMatrix(similar(M.R, element_type))
```
which was enough for all needs I had so far (in ProbNumDiffEq.jl). But then I looked up what `similar(::Array)` does, and found this: https://github.com/JuliaLang/julia/blob/c92ab5e79ea6c91d9f58c35c72f8bff2f81a9a45/base/array.jl#L378.
But then I had trouble with the `dims` argument and removed it again. 

So anyways, the `where {T,N}` is not actually needed in my implementation, so I'd suggest that I change the current version to
```julia
similar(M::PSDMatrix, element_type::Type=eltype(M)) = PSDMatrix(similar(M.R, element_type))
```

Reply:
Hmm, I see.

Personally, I would indeed prefer the suggested version or the ProbNumDiffeq version, because those snippets I understand fully :)

Reply:
Changed it :+1: And just to be clear, that messy code was my mistake not your lack of understanding :sweat_smile: